api/token_refresh_api.py

```diff
@@ -41,8 +38,9 @@ def validate_token(self, token, email):
 
   # Note: we use only POST instead of GET to avoid attacks that use GETs.
   def do_post(self):
-    """Return a new XSRF token for the current user."""
+    """Refresh the session and return a new XSRF token for the current user."""
     user = self.get_current_user()
+    users.add_signed_user_info_to_session(user.email())
```

Comment:
Is user always signed up when this function is called? Can user be None?

Reply:
Yes, the user will always be signed in at this point.  It will not be None.

If the user is not signed in, the code in basehandlers.py will generate a 403 response before this do_post() method is called.